PacMan/src/gameThreads/Game.java

```diff
@@ -103,26 +133,97 @@ private void buildMaze() {
 		
 		board.drawMaze(sprites);
 	}
+	private void removeallsprites() {
+		List<Sprite> sprites = new ArrayList<Sprite>();
+		List<Entity> entities = entityManager.getAllEntitiesPosessingComponentOfClass(GraphicsComponent.class.getName());
+		for (Entity entity : entities) {
+			GraphicsComponent graphic = (GraphicsComponent) entityManager.getComponentOfClass(GraphicsComponent.class.getName(), entity);
+			sprites.add(graphic.getSprite());	
+		}
+		board.Removesprites(sprites);
+		sprites.clear();
+	}
+	private void removeallspritesMoving() {
+		List<Sprite> sprites = new ArrayList<Sprite>();
+		List<Entity> entities = entityManager.getAllEntitiesPosessingComponentOfClass(MoveComponent.class.getName());
+		for (Entity entity : entities) {
+			GraphicsComponent graphic = (GraphicsComponent) entityManager.getComponentOfClass(GraphicsComponent.class.getName(), entity);
+			sprites.add(graphic.getSprite());	
+		}
+		board.Removesprites(sprites);
+	    sprites.clear();
+	}
+	private void AddAllspritesMoving() {
+		List<Sprite> sprites = new ArrayList<Sprite>();
+		List<Entity> entities = entityManager.getAllEntitiesPosessingComponentOfClass(MoveComponent.class.getName());
+		for (Entity entity : entities) {
+			GraphicsComponent graphic = (GraphicsComponent) entityManager.getComponentOfClass(GraphicsComponent.class.getName(), entity);
+			sprites.add(graphic.getSprite());	
+		}
+		board.addSpritesMoving(sprites);
+	    
+	}
+
+	public void NextLevel() {
+		List<Entity> entities = entityManager.getAllEntitiesPosessingComponentOfClass(GraphicsComponent.class.getName());
+		
+		for (Entity entity : entities) {
+			PhysicsComponent physic = (PhysicsComponent) entityManager
+					.getComponentOfClass(PhysicsComponent.class.getName(), entity);
+			GraphicsComponent graphic = (GraphicsComponent) entityManager
+					.getComponentOfClass(GraphicsComponent.class.getName(), entity);
+			MoveComponent move = (MoveComponent) entityManager.getComponentOfClass(MoveComponent.class.getName(),
+					entity);
+			AudioComponent audio = (AudioComponent) entityManager.getComponentOfClass(AudioComponent.class.getName(),
+					entity);
+			ScoreComponent scores = (ScoreComponent) entityManager.getComponentOfClass(ScoreComponent.class.getName(),
+					entity);
+			LifeComponent life = (LifeComponent) entityManager.getComponentOfClass(LifeComponent.class.getName(),
+					entity);
+			UserInputComponent user = (UserInputComponent) entityManager
+					.getComponentOfClass(UserInputComponent.class.getName(), entity);
+			if (graphic != null && physic != null && move == null && audio == null && scores == null && life == null && user == null) {
+				return;
+			}
+		}
+		
+      if(!board.isLevelPassed()) {
+		board.refreshlevel(++level);
+		board.setLevelPassed(false);
+		removeallsprites();
+		try {
+			map = MazeFactory.BuildMaze(entityManager);
+
+		} catch (Exception e) {
+			LOGGER.log(Level.SEVERE, e.getMessage());
+		}
+		buildMaze();
+		removeallspritesMoving();
```

Comment:
Pourquoi tu les enlève et les rajoute tout de suite après

Reply:
si  j'ai pas fais ça  le spritesMoving au de là  les Gums

Reply:
meme si tu les respawn? il ne sont pourtant pas aux memes cases

Reply:
oui   remove board ..> remove fantome from pane of maze> build maze  -->add fantome to pane of  maze